packages/date-time/src/components/CalendarDayGrid/CalendarDayGrid.base.tsx

```diff
@@ -778,4 +764,27 @@ export class CalendarDayGridBase extends BaseComponent<ICalendarDayGridProps, IC
       });
     };
   };
+
+  /**
+   * When given work week, if the days are non-contiguous, the hover states look really weird. So for non-contiguous work weeks,
+   * we'll just show week view instead
+   */
+  private getDateRangeTypeToUse = (dateRangeType: DateRangeType, workWeekDays: DayOfWeek[] | undefined): DateRangeType => {
+    if (workWeekDays && dateRangeType === DateRangeType.WorkWeek) {
+      const sortedWWDays = workWeekDays.sort();
```

Comment:
Might want to consider `slice()` before `sort()`, as `sort()` will modify `workWeekDays` in place.